File: app/src/main/java/com/boardgamegeek/sorter/PlayCountDescendingSorter.java
package com.boardgamegeek.sorter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import com.boardgamegeek.R;

public class PlayCountDescendingSorter extends PlayCountSorter {
	public PlayCountDescendingSorter(@NonNull Context context) {
		super(context);
	}

	@Override
	protected boolean isSortDescending() {
		return true;
	}

	@StringRes
	@Override
	protected int getDescriptionId() {
		return R.string.collection_sort_play_count_desc;
	}

	@StringRes
	@Override
	public int getTypeResource() {
		return R.string.collection_sort_type_play_count_desc;
	}
}

File: app/src/main/java/com/boardgamegeek/sorter/PlayCountDescendingSorter.kt
package com.boardgamegeek.sorter

import android.content.Context
import androidx.annotation.StringRes

import com.boardgamegeek.R

class PlayCountDescendingSorter(context: Context) : PlayCountSorter(context) {
    @StringRes
    override val descriptionResId = R.string.collection_sort_play_count_desc

    @StringRes
    public override val typeResId = R.string.collection_sort_type_play_count_desc

    override val isSortDescending = true
}
